### ISSUE TITLE
More explicit startup options

### ISSUE THREAD

**Issue by jrieken** [OmniSharp/omnisharp-roslyn#176]
Today OmniSharp starts on either a folder or a sln file and both project systems (msbuild, asp) do things. It sort of goes like this:

* if its a solution file
 * msbuild loads the solution unless there is a sibling unity solution
 * asp discovers all project.json files (honoring a discovery setting) and treats them as project
* if its a folder
 * msbuild checks if there is exactly one solution file (plus unity special) in that folder at the root level
 * asp again recursively check for project.json files

My proposal would be have a more explicit start behaviour:

* start with ```project.json```-file
 * asp takes that its only input
 * msbuild does nothing
* start with ```sln```-file
 * asp checks for xproj/kproj files
 * msbuild goes for csprojej files
* start with folder
  * current behaviour or ???

**Comment by nosami** [comment]
The only thing that needs to change here is the explicit start with project.json?

**Comment by jrieken** [comment]
yes and no. I wonder if we should stop doing folder-things when starting with a file. That is us looking for a single sln-file in ```foo/bar``` when Omnisharp was started as ```foo/bar/project.json``` and vice versa.
